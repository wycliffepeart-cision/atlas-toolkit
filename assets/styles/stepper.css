@layer atlas {
	atlas-stepper {
		display: flex;
		flex-wrap: nowrap;
	}
}
