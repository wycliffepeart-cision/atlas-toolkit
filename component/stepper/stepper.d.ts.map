{"version":3,"file":"stepper.d.ts","sourceRoot":"","sources":["../../../src/component/stepper/stepper.ts"],"names":[],"mappings":"AAGA,qBACa,OAAQ,SAAQ,WAAW;CAAG"}