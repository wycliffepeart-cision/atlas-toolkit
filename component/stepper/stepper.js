import { _ as __decorate } from '../../tslib.es6-2e0713a3.js';
import { Template } from '../../decorators/template.js';

var StepperTemplate = "<template>\n  <slot part=\"steps\"></slot>\n</template>\n";

let Stepper = class Stepper extends HTMLElement {
};
Stepper = __decorate([
    Template(StepperTemplate)
], Stepper);
if (!customElements.get('atlas-stepper'))
    customElements.define('atlas-stepper', Stepper);

export { Stepper };
