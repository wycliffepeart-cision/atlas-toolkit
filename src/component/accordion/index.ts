import { Accordion } from './accordion';

if (!customElements.get('atlas-accordion')) customElements.define('atlas-accordion', Accordion);
