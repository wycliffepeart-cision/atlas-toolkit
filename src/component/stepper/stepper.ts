import StepperTemplate from './stepper.html';
import { Template } from '../../decorators/template';

@Template(StepperTemplate)
export class Stepper extends HTMLElement {}
