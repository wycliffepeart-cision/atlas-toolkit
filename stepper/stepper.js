import { _ as __decorate } from '../tslib.es6-806de115.js';
import { Template } from '../decorators/template.js';

var StepperTemplate = "<template>\n  <slot part=\"steps\"></slot>\n</template>\n";

let Stepper = class Stepper extends HTMLElement {
};
Stepper = __decorate([
    Template(StepperTemplate)
], Stepper);

export { Stepper };
