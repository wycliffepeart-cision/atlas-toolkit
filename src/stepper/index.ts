import { Stepper } from './stepper'

if (!customElements.get('atlas-stepper')) customElements.define('atlas-stepper', Stepper)