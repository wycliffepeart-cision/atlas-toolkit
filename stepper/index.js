import { Stepper } from './stepper.js';
import '../tslib.es6-806de115.js';
import '../decorators/template.js';

if (!customElements.get('atlas-stepper'))
    customElements.define('atlas-stepper', Stepper);
