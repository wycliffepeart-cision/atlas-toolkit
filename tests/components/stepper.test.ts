import '../../src/component/stepper';

const STEPPER_TAG = 'atlas-stepper';

afterEach(() => {
  const elem = document.querySelector('body > *:first-child');
  elem?.remove();
});

describe('Stepper', () => {
  it('is correctly appended in the DOM', () => {
    const elem = document.createElement(STEPPER_TAG);
    document.body.appendChild(elem);

    (expect(elem) as any).toBeInTheDocument();
  });

  it('is matches snapshot', () => {
    const elem = document.createElement(STEPPER_TAG);
    document.body.appendChild(elem);

    (expect(elem.shadowRoot.innerHTML) as any).toMatchSnapshot();
  });
});
