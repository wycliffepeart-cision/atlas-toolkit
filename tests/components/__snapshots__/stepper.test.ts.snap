// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper is matches snapshot 1`] = `
<slot part="steps">
</slot>
`;
